### ISSUE TITLE
False positive on Lint/UselessAssignment

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#11124]
Given this code
```ruby
        item_fq = label_source.respond_to?(:fq) ? label_source.fq : {}
        item_fq = item_fq.symbolize_keys
        label_value = facet_config.pivot.map(&:to_sym).map { |k| item_fq[k] }
```        

I see these offenses:
```
Offenses:

app/presenters/blacklight/facet_item_pivot_presenter.rb:26:9: W: Lint/UselessAssignment: Useless assignment to variable - item_fq.
        item_fq = item_fq.symbolize_keys
        ^^^^^^^
```

But it is clearly used on the very next line.

Furthermore, when I try to disable this cop for this line

```ruby
        # rubocop:disable Lint/UselessAssignment
        item_fq = item_fq.symbolize_keys
        # rubocop:enable Lint/UselessAssignement
```        

I get the following error:
```
Offenses:

app/presenters/blacklight/facet_item_pivot_presenter.rb:27:1: W: Lint/MissingCopEnableDirective: Re-enable Lint/UselessAssignment cop with # rubocop:enable after disabling it.
        # rubocop:disable Lint/UselessAssignment
^
app/presenters/blacklight/facet_item_pivot_presenter.rb:27:9: W: [Correctable] Lint/RedundantCopDisableDirective: Unnecessary disabling of Lint/UselessAssignment.
        # rubocop:disable Lint/UselessAssignment
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
app/presenters/blacklight/facet_item_pivot_presenter.rb:29:26: W: [Correctable] Lint/RedundantCopEnableDirective: Unnecessary enabling of Lint/UselessAssignement.
        # rubocop:enable Lint/UselessAssignement
                         ^^^^^^^^^^^^^^^^^^^^^^^
```

Using rubocop 1.37.1

**Comment by tdeo** [comment]
Hello @jcoyne , I wanted to give this a look but I cannot reproduce either with 0.37.1 or master, I tried with the following file:
```ruby
item_fq = label_source.respond_to?(:fq) ? label_source.fq : {}
item_fq = item_fq.symbolize_keys
label_value = facet_config.pivot.map(&:to_sym).map { |k| item_fq[k] }
```

and the only offenses I get are:
```
example.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
item_fq = label_source.respond_to?(:fq) ? label_source.fq : {}
^
example.rb:3:1: W: Lint/UselessAssignment: Useless assignment to variable - label_value. Did you mean label_source?
label_value = facet_config.pivot.map(&:to_sym).map { |k| item_fq[k] }
^^^^^^^^^^^
```

Could you share a reproducible example for this issue?

**Comment by jcoyne** [comment]
This cleared up with one of the  recent releases. I have no further problems. Thank you.
